test/test-signal.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// Testcase to produce NodeReport on signal
+if (process.argv[2] === 'child') {
+  require('../').setEvents("signal");
+
+  function myLoop() {
+  var list = [];
+    for (var i=0; i<10000000000; i++) {
```

Comment:
Yes, this puts the test application into a busy loop, and we are testing that we can get a NodeReport using a signal in that scenario. 

The other key scenario for a signal test is an application in idle/waiting, it would be good to have a test for that at some point.